cmake/cmake_findExternalLibs.cmake

```diff
@@ -169,7 +169,7 @@ if (WITH_HDF5)
   if(MSVC)
     set(HDF5_USE_STATIC_LIBRARIES ON)
   endif()
-  find_package(HDF5 MODULE REQUIRED COMPONENTS C CXX)
+  find_package(HDF5 MODULE REQUIRED COMPONENTS CXX)
```

Comment:
Is it still possible to pass compatibility macros here?

Reply:
You mean for 1.10 vs 1.14 compat?
Yes maybe something like target_add_compile_definitions(HDF5::HDF5 INTERFACE HDF5_COMPAT)

Reply:
```
To compile an application built with a version of HDF5 that includes deprecated symbols (the default), specify: -DH5_USE_110_API (autotools) or –DH5_USE_110_API:BOOL=ON (CMake)
To build an HDF5 library with the 1.10 APIs specify --with-default-api-version=v110 (Autotools) or -DDEFAULT_API_VERSION:STRING=v110 (CMake).
```
ah ok something like:
target_add_compile_definitions(HDF5::HDF5 INTERFACE H5_USE_110_API) wizkd then probably work

Reply:
Yes, I am not sure what "the default" means in "To compile an application built with a version of HDF5 that includes deprecated symbols (the default)" but the idea is correct.
The name of the function is [target_compile_definitions](https://cmake.org/cmake/help/latest/command/target_compile_definitions.html#command:target_compile_definitions). I wonder if this ADDS to the definitions or replaces them.. Cmake docs.. Not sure if some compile definitions are already set internally by the find module. Probably not.

Reply:
a nice... seems to work